### ISSUE TITLE
Issue using NumberPrompt

### ISSUE THREAD

**Issue by garypretty** [Microsoft/botbuilder-dotnet#594]
Hi.  I am trying to use a NumberPrompt in a dialog set, but it is throwing an error when the bot is initializing.

```cs 
        public WaterUtilityBot()
        {
            dialogs = new DialogSet();
            dialogs.Add("None", new WaterfallStep[] { DefaultDialog });
            dialogs.Add("ReportLeak", new WaterfallStep[] { AskLeakDescription, AskLeakDuration, AskOkToContact, AskUpdates, AskUpdateContactInfo, SubmitLeakDetails });
            dialogs.Add("LeakDescriptionPrompt", new TextPrompt());
            dialogs.Add("LeakDurationPrompt", new TextPrompt());
            dialogs.Add("ContactNumberPrompt", new TextPrompt());
            dialogs.Add("OkToContactPrompt", new ConfirmPrompt(Culture.English));
            dialogs.Add("UpdatesPrompt", new ConfirmPrompt(Culture.English));
            dialogs.Add("GetNextPaymentDetails", new WaterfallStep[] { PaymentDetailsDialog });
            dialogs.Add("AboutMyWater", new WaterfallStep[] { AboutMyWaterDialog });
            dialogs.Add("ReadingPrompt", new TextPrompt());
            **dialogs.Add("ReadingPrompt2", new NumberPrompt<int>(Culture.English));**
            dialogs.Add("ReadingConfirmPrompt", new ConfirmPrompt(Culture.English));
        }
```

The error is occurring when the prompt dialog is added to the set in the bold line above.  I am not using this prompt at all and if I remove the above line the bot works as expected.

The error I get is as follows....

"Did not find an IBot service via the dependency resolver. Please make sure you have registered your bot with your dependency injection container."

Any ideas?

cc @cleemullins @drub0y just cc'ing you guys as I was hoping to demo this tomorrow, so if there are any issues you guys are aware of it would be good to know. Thanks!

**Comment by drub0y** [comment]
Ok, so, actually... I just realized what could be happening here and we should do a better job with logging in this case at least at the integration layer.

So, ignore what I just said about the container stuff. What's happening is that the handler is attempting to resolve the bot from the underlying `IDependencyResolver` and something is going wrong during the construction of that bot and the current handler implementation just swallows the exception and acts as if it couldn't find the bot. I'm going to take that as a separate bug to improve the exception reporting there.

Now, to figure out your exact issue, can you turn on first chance exceptions in the debugger and paste us the full details of the exception you're getting? That will help get to the root cause that's being hidden right now.

**Comment by garypretty** [comment]
@drub0y Thanks. I will check this out later and report back 👍

**Comment by garypretty** [comment]
@drub0y Ok, so this may be something funky with my local solution. The PR worked, so I am now getting a more detailed error - so thanks for that.  The error is that the DLL for Microsoft.Recognizers.Number cannot be found.  I am referencing the local Prompts project, which has the NuGet package for this installed, so I would have though that would be ok as my local bot project doesn't reference this directly. If I install the NuGet package for Recognizer.Number locally it works.  Probably some sort of mixup given that I am referencing local projects, which then in turn have NuGet packages installed??

**Comment by drub0y** [comment]
Yeah, if you're going to work against the source directly, you're bound to run into issues. It's definitely a good idea to work against 100% packages if you can. It will guarantee all the right dependencies are brought in and even help you deal with any version conflicts (e.g. getting your binding redirects right) if there are any.

I know @cleemullins is directing the team to put effort towards the packaging situation and doing things like nightly builds so, once that's done, you won't have to work like this for much longer unless you're actually working on the source itself. 

Even if you do want to work with the latest source, 'cause maybe testing a brand new fix or something, I would suggest you put your repro into its own solution and do local package references. This is still a pain right now because #391 is still pending, so you have to reference the packages via a feed pointed to each of their own output dirs vs. a common artifact dir... unless you manually run the build from the command line and specify `/p:PackageOutputPath=<someplace>` which is what the `BuildPackages.cmd` (committed yesterday) does for now until #391 is hopefully fully realized.

**Comment by garypretty** [comment]
@drub0y thanks. Yeah I'm working in the repo with the source projects as I'm building a bot with the new alexa adapter and testing / tweaking it as I go. Great advice though so thanks.
